onnxruntime/core/providers/cuda/tensor/quantize_linear.cu

```diff
@@ -226,6 +353,35 @@
   return Status::OK();
 }
 
+template <class OutT, class InT>
+Status CudaQuantizeLinearBlockStdInt4(cudaStream_t stream, const InT* input, OutT* output, const InT* scale,
+                                      const OutT* zero_point, size_t num_of_element, size_t K, size_t N,
+                                      size_t block_size) {
+  if (num_of_element <= 0)
+    return Status::OK();
+
+  CUDA_KERNEL_ASSERT((GridDim::maxElementsPerThread & 1) == 0);
```

Comment:
>CUDA_KERNEL_ASSERT [](http://example.com/codeflow?start=2&length=18)

CUDA_KERNEL_ASSERT is supposed to use inside a kernel. you can just assume maxElementsPerThread is a multiplier of 2

[http://docs.nvidia.com/cuda/cuda-c-programming-guide/#assertion](http://docs.nvidia.com/cuda/cuda-c-programming-guide/#assertion)

Reply:
changed to static_assert